pkg/apis/batch/validation/validation.go

```diff
@@ -236,26 +236,32 @@ func validatePodFailurePolicy(spec *batch.JobSpec, fldPath *field.Path, allErrs
 			}
 			if rule.OnExitCodes.ContainerName != nil && !containerNames.Has(*rule.OnExitCodes.ContainerName) {
 				path := onExitCodesPath.Child("containerName")
-				allErrs = append(allErrs, field.Invalid(path, *rule.OnExitCodes.ContainerName, "must be one of the container names in the pod template"))
+				allErrs = append(allErrs, field.Invalid(path, *rule.OnExitCodes.ContainerName, "must be one of the container or initContainer names in the pod template"))
 			}
 			valuesPath := onExitCodesPath.Child("values")
 			if len(rule.OnExitCodes.Values) == 0 {
 				allErrs = append(allErrs, field.Invalid(valuesPath, rule.OnExitCodes.Values, "at least one value is required"))
 			} else {
 				isOrdered := true
 				uniqueValues := sets.NewInt32()
+				duplicatedValues := sets.NewInt32()
 				for j, exitCodeValue := range rule.OnExitCodes.Values {
 					if rule.OnExitCodes.Operator == batch.PodFailurePolicyOnExitCodesOpIn && exitCodeValue == 0 {
 						valuePath := valuesPath.Index(j)
 						allErrs = append(allErrs, field.Invalid(valuePath, exitCodeValue, "must not be 0 for the In operator"))
 					}
-					uniqueValues.Insert(rule.OnExitCodes.Values[j])
-					if j > 0 && rule.OnExitCodes.Values[j-1] > rule.OnExitCodes.Values[j] {
+					if uniqueValues.Has(exitCodeValue) {
+						duplicatedValues.Insert(exitCodeValue)
```

Comment:
Im ok either way, but then we will have a separate error per duplicated value, and the index we report is arbitrary. in general was thinking about duplicates more that it is a property of a list to contain duplicates rather than of individual elements. if we report error per element, should we apply the same strategy for checking if the list is ordered. Maybe we could just report elements that are smaller strictly than the previous element  this way we can both handle ordering and duplicates?

Reply:
For ordered it makes sense to use the entire list. For duplicated it might be useful to know the position of each repeated index, and it's simpler to do that using the a different error.

A bit unrelated: let's add an upper limit for the maximum number of exit codes you can use. How about 255?

Reply:
I see, in order to constrain the size of the podFailurePolicy object I introduced the following limits:
```golang
	// maximal number of rules in pod failure policy
	maxPodFailurePolicyRules = 100

	// maximal number of values for the OnExitCodes requirement in pod failure policy
	maxPodFailurePolicyOnExitCodesValues = 255

	// maximal number of patterns for the OnPodConditions requirement in pod failure policy
	maxPodFailurePolicyOnPodConditionsPatterns = 100
```

Reply:
Discussed offline: 20 rules times 20 conditions, with about 100 characters per conditionType, leads to about 40kb.

Reply:
Updated accordingly